embedded-graphics/src/geometry/size.rs

```diff
@@ -0,0 +1,244 @@
+use crate::geometry::Point;
+use core::ops::{Add, AddAssign, Index, Sub, SubAssign};
+
+/// 2D size.
+///
+/// `Size` is used to define the width and height of an object.
+///
+/// [Nalgebra] support can be enabled with the `nalgebra_support` feature. This implements
+/// `From<Vector2<N>>` and `From<&Vector2<N>>` where `N` is `Scalar + Into<u32>`. This allows use
+/// of Nalgebra's [`Vector2`] with embedded-graphics where `u32`, `u16` or `u8` is used for value
+/// storage.
+///
+/// # Examples
+///
+/// ## Create a `Size` from two integers
+///
+///
+/// ```rust
+/// use embedded_graphics::geometry::{size, Size};
+///
+/// // Create a size using the `new` constructor method
+/// let c1 = Size::new(10, 20);
+///
+/// // Create a size using the handy `size` function
+/// let c2 = size(10, 20);
+///
+/// assert_eq!(c1, c2);
+/// ```
+///
+/// ## Create a `Size` from a Nalgebra `Vector2`
+///
+/// _Be sure to enable the `nalgebra_support` feature to get [Nalgebra] integration._
+///
+/// Any `Vector2<N>` can be used where `N: Into<u32> + nalgebra::Scalar`. This includes the primitive types `u32`, `u16` and `u8`.
+///
+/// ```rust
+/// # #[cfg(feature = "nalgebra_support")] {
+/// use nalgebra::Vector2;
+/// use embedded_graphics::geometry::Size;
+///
+/// assert_eq!(Size::from(Vector2::new(10u32, 20)), Size::new(10u32, 20));
+/// assert_eq!(Size::from(Vector2::new(10u16, 20)), Size::new(10u32, 20));
+/// assert_eq!(Size::from(Vector2::new(10u8, 20)), Size::new(10u32, 20));
+/// # }
+/// ```
+///
+/// `.into()` can also be used, but may require more type annotations:
+///
+/// ```rust
+/// # #[cfg(feature = "nalgebra_support")] {
+/// use nalgebra::Vector2;
+/// use embedded_graphics::geometry::Size;
+///
+/// let c: Size = Vector2::new(10u32, 20).into();
+///
+/// assert_eq!(c, Size::new(10u32, 20));
+/// # }
+/// ```
+///
+/// [`Drawable`]: ../drawable/trait.Drawable.html
+/// [`Point`]: struct.Point.html
+/// [`Vector2<N>`]: https://docs.rs/nalgebra/0.18.0/nalgebra/base/type.Vector2.html
+/// [`Vector2`]: https://docs.rs/nalgebra/0.18.0/nalgebra/base/type.Vector2.html
+/// [Nalgebra]: https://docs.rs/nalgebra
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub struct Size {
+    /// The width.
+    pub width: u32,
+
+    /// The height.
+    pub height: u32,
+}
+
+impl Size {
+    /// Create a new coordinate with X and Y values
+    pub const fn new(width: u32, height: u32) -> Self {
+        Size { width, height }
+    }
+
+    /// Creates a size from two corner points of a bounding box.
+    ///
+    /// TODO: issue #142
+    pub fn from_bounding_box(corner_1: Point, corner_2: Point) -> Self {
```

Comment:
Naming suggestion: `from_points` or is that too abstract? `from_corners`? 😬 

```suggestion
    pub fn from_points(corner_1: Point, corner_2: Point) -> Self {
```

Reply:
I would really like to remove this completely when we've found solutions for #142 and `Rect(angle)`. Should I leave the temporary name for now or change it?

Reply:
Ok, in that case leave the name 🙂. Could be worth adding a `#[deprecated]` attribute so it's harder to forget that this is left in.